code/datums/world_topic.dm

```diff
@@ -26,7 +26,7 @@
 	var/require_comms_key = FALSE
 
 /datum/world_topic/proc/TryRun(list/input)
-	key_valid = config && (CONFIG_GET(string/comms_key) == input["key"])
+	key_valid = (CONFIG_GET(string/comms_key) == input["key"]) && CONFIG_GET(string/comms_key) && input["key"]
```

Comment:
With this the key isn't being compared anymore? You nee to make sure the key isn't length 0 or otherwise null AND compare the configured key to the key passed in input.

Reply:
The key being valid is checked inside parenthesis. 
It could be split out however it's functional as is